compiler/src/dmd/dtemplate.d

```diff
@@ -5625,7 +5625,11 @@ extern (C++) final class TemplateValueParameter : TemplateParameter
         if (e)
         {
             e = e.syntaxCopy();
-            if ((e = e.expressionSemantic(sc)) is null)
+            const saveInDefaultArg = sc.inDefaultArg;
+            sc.inDefaultArg = true;
```

Comment:
This 'save and restore' pattern is risky, since `sc` may be referenced by siblings. It's safer to push and pop a scope, see: https://github.com/dlang/dmd/pull/15434#discussion_r1270170347

Although in these cases, I don't think it's possible for there to be expression siblings that aren't default args.

Reply:
> It's safer to push and pop a scope

Welp, that aged well :laughing:

https://github.com/dlang/dmd/pull/16781